template/src/main.ts

```diff
@@ -2,17 +2,18 @@ import Vue from 'vue';
 import VueRouter from 'vue-router';
 
 import './sass/main.scss';
+import {NavbarComponent} from './components/navbar';
 
-import { HomeComponent } from './components/home';
-import { AboutComponent } from './components/about';
-import { ListComponent } from './components/list';
-import { NavbarComponent } from './components/navbar';
+import {createRoutes} from './module';
 
-import { createRouter } from './router';
+
+// register the plugin
+Vue.use(VueRouter);
```

Comment:
I would prefer to leave this in the `router.ts` file and move the `createRoutes` function to a `routes.ts` file

Reply:
it conflicts the npm-module option, because when you're building a module, you want to export the routings array, and the main component, but not to initiate it yourself.
the initiator will be the main application that uses that module.

Reply:
I understand. I was thinking of leaving the initialisation of the router in the `router.ts` and that imports the `createRoutes` function from `routes.ts`. The `module.ts` would also import/export the `createRoutes` function from the `routes.ts` file. Does that make sense?